### PR TITLE
Reduced GCP SA permission scope

### PR DESCRIPTION
**Pull request by DeeAjayi**

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -1,10 +1,122 @@
...
+  count = var.feature_blob_storage_enabled ? 1 : 0
+
+  project = var.project
+  role    = google_project_iam_custom_role.gcs_tfy_role[count.index].id
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+
+  condition {
+    title       = "Restrict to tfy storage buckets"
+    description = "Allows access to buckets that start with 'tfy'"
+    expression  = "resource.name.startsWith('projects/_/buckets/tfy')"
```
Comment (dunefro) [inline iam.tf:79]:
```suggestion
    expression  = "resource.name.startsWith('projects/_/buckets/${module.blob_storage[0].name}')"
```

File: iam.tf [hunk 2]
```diff
@@ -1,10 +1,122 @@
...
+  }
+}
+
+// add service account to token creator role
```
Comment (dunefro) [inline iam.tf:83]: Please change the comment respectively

File: iam.tf [hunk 3]
```diff
@@ -1,10 +1,122 @@
...
+resource "google_project_iam_member" "token_creator_role_binding" {
+  count = var.feature_blob_storage_enabled ? 1 : 0
+
+  project = var.project
+  role    = "roles/iam.serviceAccountTokenCreator"
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+}
+
+// add artifact registry role to service account
```
Comment (dunefro) [inline iam.tf:92]: Please change the comment respectively

File: iam.tf [hunk 4]
```diff
@@ -1,10 +1,122 @@
...
+resource "google_project_iam_member" "artifact_registry_role_binding" {
+  count = var.feature_docker_registry_enabled ? 1 : 0
+
+  project = var.project
+  role    = "roles/artifactregistry.admin"
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+}
+
+// add container cluster viewer role to service account
```
Comment (dunefro) [inline iam.tf:101]: Please change the comment respectively

File: iam.tf [hunk 5]
```diff
@@ -1,10 +1,122 @@
...
+resource "google_project_iam_member" "container_cluster_viewer_role_binding" {
+  count = var.feature_cloud_integration_enabled ? 1 : 0
+
+  project = var.project
+  role    = "roles/container.clusterViewer"
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+}
+
+// add container viewer role to service account
```
Comment (dunefro) [inline iam.tf:110]: Please change the comment respectively

File: iam.tf [hunk 6]
```diff
@@ -1,10 +1,122 @@
...
+resource "google_project_iam_member" "container_viewer_role_binding" {
+  count = var.feature_cloud_integration_enabled ? 1 : 0
+
+  project = var.project
+  role    = "roles/container.viewer"
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+}
+
+// service account key
```
Comment (dunefro) [inline iam.tf:119]: Please change the comment respectively

File: iam.tf [hunk 7]
```diff
@@ -1,10 +1,122 @@
...
+
+  condition {
+    title       = "Restrict to tfy storage buckets"
+    description = "Allows access to buckets that start with 'tfy'"
+    expression  = "resource.name.startsWith('projects/_/buckets/tfy')"
+  }
+}
+
+// add service account to token creator role
+resource "google_project_iam_member" "token_creator_role_binding" {
```
Comment (dunefro) [inline iam.tf:84]: Change the terraform resource name accordingly

**Review by dunefro — CHANGES_REQUESTED** [review]

File: iam.tf [hunk 8]
```diff
@@ -1,10 +1,122 @@
...
+  project      = var.project
+  display_name = "Terraform-managed truefoundry platform service account"
+  description = "Truefoundry platform user with access to artifact registry, blob storage and secrets manager"
+}
+
+// custom roles for secret manager
+resource "google_project_iam_custom_role" "secret_manager_tfy_role" {
+  count       = var.feature_secrets_enabled ? 1 : 0
+
+  role_id     = replace("${local.trufoundry_platform_resources}_bucket_secret_manager_tfy_role", "-", "_")
```
[the remark this answers is not in the document]
Comment (DeeAjayi) [reply iam.tf:13]: hyphens are not accepted

File: iam.tf [hunk 9]
```diff
@@ -1,10 +1,122 @@
...
+    "secretmanager.versions.access",
+    "resourcemanager.projects.get",
+  ]
+}
+
+// custom role for GCS
+resource "google_project_iam_custom_role" "gcs_tfy_role" {
+  count       = var.feature_blob_storage_enabled ? 1 : 0
+
+  role_id     = replace("${local.trufoundry_platform_resources}_bucket_gcs_tfy_role", "-", "_")
```
[the remark this answers is not in the document]
Comment (DeeAjayi) [reply iam.tf:31]: no we can't 

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_iam_custom_role#role_id

**Review by dunefro — APPROVED** [review]
LGTM

**Pull request merged**